### PR TITLE
[W5][T09-1] Yeo Cheng Hong

### PR DESCRIPTION
**Pull request by chyeo**
Implemented a 'sort' command in the addressbook to list the people in capital [A-Z] followed by lower capital [a-z] order

### CODE REVIEW

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -74,4 +78,16 @@ public boolean equals(Object other) {
                 || (other instanceof AddressBook // instanceof handles nulls
                         && this.allPersons.equals(((AddressBook) other).allPersons));
     }
+
+    public List<ReadOnlyPerson> sortPersons( List<ReadOnlyPerson> allPersons) {
+        if (allPersons.size()>1) {
```
Comment (lycjackie) [inline src/seedu/addressbook/data/AddressBook.java]: There should be a space between `allPerson.size()` ,`>` and `1`

Refer to [CS2113T Java Coding Standard](https://nuscs2113-ay1819s2.github.io/website/coding-standards/java/intermediate.html) - White space within a statement section.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package seedu.addressbook.commands;
 
+import seedu.addressbook.data.AddressBook;
 import seedu.addressbook.data.person.ReadOnlyPerson;
 
 import java.util.List;
 
-
```
Comment (lycjackie) [inline src/seedu/addressbook/commands/SortCommand.java]: Try not to have unnecessary changes of new line for commits

File: src/seedu/addressbook/data/AddressBook.java [hunk 3]
```diff
@@ -87,7 +87,6 @@ public int compare(ReadOnlyPerson a, ReadOnlyPerson b) {
                 }
             });
         }
-
```
Comment (lycjackie) [inline src/seedu/addressbook/data/AddressBook.java]: Try not to have unnecessary changes in new line for commit

**Comment by lycjackie** [comment]
Overall, really get work done to make small incremental changes to complete one feature! 👍

Like that you mention how the sorting is based from `[A-Z]` follow by `[a-z]`

For the code below, might want to consider putting the sorting logic inside `UniquePersonList` class.
This provide more flexibilty in the code logic if there's any changes in the future without touching the `AddressBook` class
https://github.com/nusCS2113-AY1819S2/addressbook-level2/blob/623f8579c3808fe8777e2c1a9d6e0fa1b75dafb8/src/seedu/addressbook/data/AddressBook.java#L82-L89

**Comment by chyeo** [comment]
Greetings @lycjackie,
Thank you very much your valuable feedback!
Have updated the coding standard accordingly and will work on putting the sorting logic in the `UniquePersonList` class.

**Comment by WendyBaiYunwei** [comment]
Good Job! But you may try to add the sort function to the help section as well?

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -72,6 +72,12 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== List all persons in sorted order : `sort`
+
+Shows a list of all persons, along with their non-private details, in the address book. +
```
Comment (RubaP) [inline docs/UserGuide.adoc:77]: Good job on updating the UserGuide as well.

File: test/expected.txt [hunk 5]
```diff
@@ -297,6 +297,11 @@
 || 
 || 0 persons listed!
 || ===================================================
+|| ===================================================
+|| Enter command: || [Command entered:  sort]
+||
+|| 0 persons listed!
+|| ===================================================
```
Comment (RubaP) [inline test/expected.txt:304]: Good that you have attempted to update the I/O test cases. However, your test case only checks the empty list sort scenario. Try to add more meaningful test cases in the future. Also, I would suggest adding JUnit test cases as well.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+
+/**
+ * Lists all persons in the address book to the user in a sorted order from capital [A-Z] to lower capital [a-z].
```
Comment (RubaP) [inline src/seedu/addressbook/commands/SortCommand.java:9]: Good that you have added meaning full header comment for the new Class.

File: src/seedu/addressbook/data/AddressBook.java [hunk 7]
```diff
@@ -74,4 +78,15 @@ public boolean equals(Object other) {
                 || (other instanceof AddressBook // instanceof handles nulls
                         && this.allPersons.equals(((AddressBook) other).allPersons));
     }
+
+    public List<ReadOnlyPerson> sortPersons( List<ReadOnlyPerson> allPersons) {
+        if (allPersons.size() > 1) {
+            Collections.sort(allPersons, new Comparator<ReadOnlyPerson>() {
+                public int compare(ReadOnlyPerson a, ReadOnlyPerson b) {
+                    return a.getName().compareTo(b.getName());
```
Comment (RubaP) [inline src/seedu/addressbook/data/AddressBook.java:86]: You are comparing two Name objects using the compareTo method. However, the comapreTo implementation in Name Class is not available. This seems a compilation error.

**Review by RubaP — COMMENTED** [review]
Good attempt. Always add test cases as a practice whenever you make a functional change to avoid regression errors in the future. Please read the review comments and close the PR.
